ssl/record/rec_layer_s3.c

```diff
@@ -186,11 +186,15 @@ int ssl3_read_n(SSL *s, size_t n, size_t max, int extend, int clearold,
         return 0;
 
     rb = &s->rlayer.rbuf;
-    if (rb->buf == NULL)
+    if (rb->buf == NULL) {
         if (!ssl3_setup_read_buffer(s)) {
             /* SSLfatal() already called */
             return -1;
         }
+        if (rb->buf == NULL)
```

Comment:
This is surely a false positive from coverity, i.e. it doesn't know that a success return from `ssl3_setup_read_buffer` means that `rb->buf != NULL`. I would remove this change and mark it as a false positive.

Reply:
Okay, removed.